shell/browser/ui/electron_gtk.sigs

```diff
@@ -0,0 +1,6 @@
+GtkFileChooserNative* gtk_file_chooser_native_new(const gchar* title, GtkWindow* parent, GtkFileChooserAction action, const gchar* accept_label, const gchar* cancel_label);
+void gtk_native_dialog_set_modal(GtkNativeDialog* self, gboolean modal);
+void gtk_native_dialog_show(GtkNativeDialog* self);
+void gtk_native_dialog_hide(GtkNativeDialog* self);
+gint gtk_native_dialog_run(GtkNativeDialog* self);
+void gtk_native_dialog_destroy(GtkNativeDialog* self);
```

Comment:
how was this file generated? Might we have to update it in future? Perhaps a comment in the GN file would be in order?

Reply:
Signatures are manually declared here for which need function ptrs, currently it covers the optional functions for the native file chooser implementation https://github.com/electron/electron/blob/4d4682c0e38cd49add16310278726e48de9db2f5/shell/browser/ui/file_dialog_gtk.cc#L29-L38. But moving forward to support gtk4 and gtk3 at runtime, this file will have to be extended for all gtk functions we use and then perform necessary runtime checking before using the ptr.

The `.sig` file does not seem to take comment section, I will add this info to the GN file.

Reply:
Addressed in https://github.com/electron/electron/pull/33818/commits/6c9d92c41d382ba627b803ede5d7b7a153bebbb5